docs/proposals/dynamic-openshift-settings-configmaps.md

```diff
@@ -0,0 +1,79 @@
+# Changing OpenShift settings dynamically through ConfigMaps
+
+## Problem
+
+Currently, there are settings contained within the master-config.yaml file that
+are of value to an unpriviledged user. Specifically, the `autoProvisionEnabled`
+boolean under `jenkinsPipelineConfig` is only available to users with the
+ability to write to the master-config.
+
+## Design proposal:
+
+- In each project there can exist one or more ConfigMaps annotated with
+  'project.openshift.io/configuration'
+
+- The ConfigMap will contain the key/value pair of setting(s) to be overridden
+  (key) and the value(s) to use.
+
+- Some cluster settings will be overridable at the project level using this
+  method, and other project specific settings can be introduced via the
+  ConfigMap as well.
+
+- For values with more than one entry, e.g. `corsAllowedOrigins`, yaml or json
+  array notation could be used.
+
+- (Optional) Have a utility function for consolidating settings over multiple
+  ConfigMaps. Have documentation to alert users that overlapping settings causes
+  nondeterministic or unsupported behavior.
+
+## Issues:
+
+- We will likely not be exposing all settings in this way, so there needs to be
+  an easy way for affected users to discover these.
+
+- Will admins be able to disable this feature, and if so, how?
+
+## Other solutions considered
+
+Handling settings like this through project annotations was explored, but
+because unpriviledged users cannot annotate their own projects, it was
```

Comment:
not sure what you mean by "we already use the namespace for settings like this".

the reason we didn't want to use namespace annotations is because we wanted lesser privileged users to be able to control these settings and (as you noted) regular users can't annotate their namespace.

Now, maybe there's a concern that this would give too much power to otherwise unprivileged users (any user who can create a configmap in your project, can control the settings of that project) but the current situation seems problematic too (my admin gives me a project and now i need to go back to him/her and ask for particular settings to be changed).  it's particularly problematic for self-provisioning.